smdebug_rulesconfig/actions/actions.py

```diff
@@ -0,0 +1,184 @@
+
+
+class StopTraining(Action):
+    def __init__(self, training_job_prefix=None):
+        """
+        Action for stopping the training job when a rule is fired. Note that a policy must be created in the AWS
+        account to allow the sagemaker role to stop the training job:
+
+        ```
+        {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Sid": "VisualEditor0",
+                    "Effect": "Allow",
+                    "Action": "sagemaker:StopTrainingJob",
+                    "Resource": "arn:aws:sagemaker:*:<account_id>:training-job/*"
+                }
+            ]
+        }
+        ```
+
+        :param training_job_prefix: The prefix of the training job to stop if the rule is fired. This must only refer
+            to one active training job, otherwise no training job will be stopped.
+        """
+        self.use_default_training_job_prefix = True
+        if training_job_prefix is not None:
+            validate_training_job_prefix("training_job_prefix", training_job_prefix)
+            self.use_default_training_job_prefix = False
+        super().__init__(training_job_prefix=training_job_prefix)
+
+    def update_training_job_prefix_if_not_specified(self, training_job_name):
+        """
+        Update the training job prefix to be the training job name if the user has not already specified a custom
+        training job prefix. This is meant to be called via the sagemaker SDK when `estimator.fit` is called by the
+        user.
+
+        :param training_job_name: Name of the training job, passed in when `estimator.fit` is called.
+        """
+        validate_training_job_prefix("training_job_name", training_job_name)
+
+        if self.use_default_training_job_prefix:
+            self.action_parameters["training_job_prefix"] = training_job_name
+
+
+class Email(Action):
+    def __init__(self, email_address):
+        """
+        Action for sending an email to the provided email address when the rule is fired. Note that a policy must be
+        created in the AWS account to allow the sagemaker role to send an email to the user:
+
+        ```
+        {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Sid": "VisualEditor0",
+                    "Effect": "Allow",
+                    "Action": [
+                        "sns:Publish",
+                        "sns:CreateTopic",
+                        "sns:Subscribe"
+                    ],
+                    "Resource": "arn:aws:sns:*:<account-id>:SMDebugRules"
+                }
+            ]
+        }
+        ```
+
+        :param email_address: Email address to send the email notification to.
+        """
+        validate_email_address("email_address", email_address)
+        super().__init__(endpoint=email_address)
+
+
+class SMS(Action):
+    def __init__(self, phone_number):
```

Comment:
Type annotate here. It isn't clear if the param is a str or int

Reply:
Done.